src/test/scala/SparkIntegrationTest.scala

```diff
@@ -180,6 +180,108 @@ class SparkIntegrationTest
     WeaviateDocker.deleteClass()
   }
 
+  test("Article with string array") {
+    WeaviateDocker.createClass(Property.builder()
+      .dataType(List[String]("string[]").asJava)
+      .name("keywords")
+      .build()
+    )
+    import spark.implicits._
+    val articles = Seq(ArticleWithStringArray("Sam", "Sam and Sam", 3, Array("yo", "hey"))).toDF
+
+    articles.write
+      .format("io.weaviate.spark.Weaviate")
+      .option("scheme", "http")
+      .option("host", "localhost:8080")
+      .option("className", "Article")
+      .mode("append")
+      .save()
+
+    val results = client.data().objectsGetter()
+      .withClassName("Article")
+      .run()
+
+    if (results.hasErrors) {
+      println("Error getting Articles" + results.getError.getMessages)
+    }
+
+    assert(results.getResult.size == 1)
+    val props = results.getResult.get(0).getProperties
+    assert(props.get("title") == "Sam")
+    assert(props.get("content") == "Sam and Sam")
+    assert(props.get("wordCount") == 3)
+    assert(props.get("keywords") == List("yo", "hey").asJava)
```

Comment:
These strings aren't even getting written because we are trying to write the unsafe array datastructure into weaviate which it doesn't like

Reply:
how is the test passing if the strings aren't written to Weaviate?